File: src/storeFx/playerContext.ts
export const audio = new Audio();
audio.crossOrigin = 'anonymous';

export const replayGainValue: { contents?: number } = {};
export const skipping = { contents: false };

export const value: {
  contents?: {
    audioContext: AudioContext;
    replayGainNode: GainNode;
    volumeNode: GainNode;
  };
} = {};

File: src/storeFx/playerContext.ml
open DomExtra

let audio = HtmlAudioElement.make ()
let () = audio |. HtmlAudioElement.setCrossOrigin `anonymous
let replayGainValue : float option ref = ref None
let skipping = ref false

type playerContext = {
  audioContext : WebAudio.AudioContext.t;
  replayGainNode : WebAudio.AudioNode.t;
  volumeNode : WebAudio.AudioNode.t;
}

let value : playerContext option ref = ref None
